File: stdlib/go/MathTools.go
package main
import (
	"math"
)
func arcsine(x2 dynamic) dynamic{
	return math.Asin(x2.(float64));
}

func arccos(x2 dynamic) dynamic{
	return math.Acos(x2.(float64));
}
	
func cosine(x3 dynamic) dynamic {
	return math.Cos(x3.(float64));
}
	
func power(x4 dynamic, y dynamic) dynamic {
	return math.Pow(x4.(float64), y.(float64));
}
	
func sine(x5 dynamic) dynamic {
	return math.Sin(x5.(float64));
}
	
func floorValue(x6 dynamic) dynamic {
	return math.Floor(x6.(float64));
}
	

File: stdlib/c/MathTools.c
#include <math.h>

double arcsine(void* x) {
    double* x2 = (double*)x;
    double x3 = *x2;
    return asin(x3);
}

double arccos(void* x) {
    double* x2 = (double*)x;
    double x3 = *x2;
    return acos(x3);
}

double cosine(void* x) {
    double* x2 = (double*)x;
    double x3 = *x2;
    return cos(x3);
}

double power(void* x, void* y) {
    double* x2 = (double*)x;
    double x3 = *x2;
    double* y2 = (double*)y;
    double y3 = *y2;
    return pow(x3, y3);
}

double sine(void* x) {
    double* x2 = (double*)x;
    double x3 = *x2;
    return sin(x3);
}

double floorValue(void* x) {
    double* x2 = (double*)x;
    double x3 = *x2;
    return floor(x3);
}